core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -457,18 +477,19 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
                localTopologyManager.confirmRebalancePhase(cacheName, cacheTopology.getTopologyId(), cacheTopology.getRebalanceId(), null);
          }
 
+         PartitionHandlingConfiguration phConfig = configuration.clustering().partitionHandling();
+         boolean wasMember = previousWriteCh != null && previousWriteCh.getMembers().contains(rpcManager.getAddress());
+         boolean deletePastMemberVals = wasMember && phConfig.whenSplit() != PartitionHandling.ALLOW_READ_WRITES && phConfig.mergePolicy() == null;
```

Comment:
Only if it is explicitly set to null in the configuration. This is so that the previous merge behaviour is still available.